sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLog.scala

```diff
@@ -325,6 +338,23 @@ class HDFSMetadataLog[T <: AnyRef : ClassTag](sparkSession: SparkSession, path:
     }
   }
 
+  /**
+   * Removes the oldest log files so that we only have the
+   * latest `minEntriesToMaintain` log files.
+   */
+  def purgeOldest(minEntriesToMaintain: Int): Unit = {
```

Comment:
This might not be safe for how the files are being written out ? lets move the purging logic out to a separate PR maybe ?